test/e2e/client_tls_test.go

```diff
@@ -473,3 +931,60 @@ func encodeKey(key *rsa.PrivateKey) string {
 		Bytes: x509.MarshalPKCS1PrivateKey(key),
 	}))
 }
+
+// curl execs a Curl command in the test client pod and returns an error value.  The Curl command uses the specified
+// certificate from the client certificates configmap, sends a request for the canary route via the router's internal
+// service. Returns the HTTP status code returned from curl, and an error either if there is an HTTP error, or if
+// there's another error in running the command. If the error was not an HTTP error, the HTTP status code returned will
+// be -1.
+func curl(t *testing.T, clientPod *corev1.Pod, certName, endpoint, ingressControllerIP string, verbose bool) (int64, error) {
```

Comment:
*nit* should the name of this function reflect that it's getting a status code? i.e. `curlHttpStatus` or `getHttpStatusCode`?

Reply:
Ehh sorry I see you didn't write this, but adapted it. Nevermind this.

Reply:
I think it's still reasonable to try to convey what's being returned. I'm not sure I love either of those names, but I'll see if I can come up with something that will work